### PR TITLE
Issue #863 transport regridding

### LINKED ISSUE #863: add support for transport regridding
Our regridding functionality is implemented and tested for flow, but is should also be able to deal with transport. 

Refinement
-------------
- This also holds for clipping and masking
- Add tests
- Complications resulting from regridded/clipped boundary conditions are out of scope

### PR DESCRIPTION
**Pull request by luitjansl**
Fixes #863

# Description
 transport models now can be regridded, 

# Checklist
- [X] Links to correct issue
- [X] Update changelog, if changes affect users
- [X] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [X] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/tests/fixtures/flow_transport_simulation_fixture.py [hunk 1]
```diff
@@ -157,7 +154,7 @@ def flow_transport_simulation():
         concentration_boundary_type="Aux",
         screen_top=[0.0, 0.0],
         screen_bottom=[-1.0, -1.0],
-        rate=[1.0, -2.0],
+        rate=[0.1, -0.2],
```
Comment (JoerivanEngelen) [inline imod/tests/fixtures/flow_transport_simulation_fixture.py:157]: Please add a comment how and why we departed from the original mf6 example

**Review by JoerivanEngelen — APPROVED** [review]
I have some suggestions, approving in advance.

File: imod/mf6/adv.py [hunk 2]
```diff
@@ -10,15 +10,18 @@
 """
 
 from copy import deepcopy
+from typing import Tuple
```
[the remark this answers is not in the document]
Comment (luitjansl) [reply imod/mf6/adv.py:13]: done

File: imod/tests/fixtures/flow_transport_simulation_fixture.py (hunk 1 above)
Comment (luitjansl) [reply imod/tests/fixtures/flow_transport_simulation_fixture.py:157]: done. note that the fixture did not mention the original mf6 example, so nothing was promised wrt that. 
I added a paragraph to explain where the model in the fixture comes from and why it was changed.

File: imod/mf6/package.py [hunk 3]
```diff
@@ -537,12 +541,15 @@ def mask(self, mask: GridDataArray) -> Any:
...
+        if hasattr(self,"auxiliary_data_fields"):
+            remove_expanded_auxiliary_variables_from_dataset(self)
         for var in self.dataset.data_vars.keys():
             if self._skip_masking_variable(var, self.dataset[var]):
                 masked[var] = self.dataset[var]
             else:
                 masked[var] = self._mask_spatial_var(var, mask)
-
+        if hasattr(self,"auxiliary_data_fields"):
+            expand_transient_auxiliary_variables(self)
```
[the remark this answers is not in the document]
Comment (luitjansl) [reply imod/mf6/package.py:552]: nice idea, created issue #909 for that.
